src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/prometheus/silence-list/silence-list.component.ts

```diff
@@ -0,0 +1,196 @@
+  selection = new CdTableSelection();
+  modalRef: BsModalRef;
+  customCss = {
+    'label label-danger': 'active',
+    'label label-warning': 'pending',
+    'label label-default': 'expired'
+  };
+  sorts: SortPropDir[] = [{ prop: 'endsAt', dir: SortDirection.desc }];
+
+  constructor(
+    private authStorageService: AuthStorageService,
+    private i18n: I18n,
+    private cdDatePipe: CdDatePipe,
+    private prometheusService: PrometheusService,
+    private modalService: BsModalService,
+    private notificationService: NotificationService,
+    private urlBuilder: URLBuilderService,
+    private actionLabels: ActionLabelsI18n,
+    private succeededLabels: SucceededActionLabelsI18n
+  ) {
+    this.permission = this.authStorageService.getPermissions().prometheus;
+  }
+
+  ngOnInit() {
+    const selectionExpired = (selection: CdTableSelection) =>
+      selection.first() && selection.first().status.state === 'expired';
+    this.tableActions = [
+      {
+        permission: 'create',
+        icon: 'fa-plus',
+        routerLink: () => this.urlBuilder.getCreate(),
+        canBePrimary: (selection: CdTableSelection) => !selection.hasSingleSelection,
+        name: this.actionLabels.CREATE
+      },
+      {
+        permission: 'create',
+        canBePrimary: (selection: CdTableSelection) =>
+          selection.hasSingleSelection && selectionExpired(selection),
+        disable: (selection: CdTableSelection) =>
+          !selection.hasSingleSelection ||
+          selection.first().cdExecuting ||
+          (selection.first().cdExecuting && selectionExpired(selection)) ||
+          !selectionExpired(selection),
+        icon: 'fa-copy',
+        routerLink: () => this.urlBuilder.getRecreate(this.selection.first().id),
+        name: this.actionLabels.RECREATE
+      },
+      {
+        permission: 'update',
+        icon: 'fa-pencil',
+        canBePrimary: (selection: CdTableSelection) =>
+          selection.hasSingleSelection && !selectionExpired(selection),
+        disable: (selection: CdTableSelection) =>
+          !selection.hasSingleSelection ||
+          selection.first().cdExecuting ||
+          (selection.first().cdExecuting && !selectionExpired(selection)) ||
+          selectionExpired(selection),
+        routerLink: () => this.urlBuilder.getEdit(this.selection.first().id),
+        name: this.actionLabels.EDIT
+      },
+      {
+        permission: 'delete',
+        icon: 'fa-trash-o',
+        canBePrimary: (selection: CdTableSelection) =>
+          selection.hasSingleSelection && !selectionExpired(selection),
+        disable: (selection: CdTableSelection) =>
+          !selection.hasSingleSelection ||
+          selection.first().cdExecuting ||
+          selectionExpired(selection),
+        click: () => this.expireSilence(),
+        name: this.actionLabels.EXPIRE
+      }
+    ];
+    this.columns = [
+      {
+        name: this.i18n('ID'),
+        prop: 'id',
+        flexGrow: 2
```

Comment:
This value is bigger then all the others, changing to 3 prevents the line to break.
```suggestion
        flexGrow: 3
```